src/libYARP_sig/src/IntrinsicParams.cpp

```diff
@@ -51,6 +52,9 @@ void IntrinsicParams::fromProperty(const yarp::os::Property& intrinsic)
     principalPointX = intrinsic.find("principalPointX").asFloat64();
     principalPointY = intrinsic.find("principalPointY").asFloat64();
 
+    // The physical focal length is optional
+    physFocalLength = intrinsic.check("physFocalLength", yarp::os::Value(0.0)).asFloat64();
+
```

Comment:
The parsing of file does not allow optional field (see @Nicogene comments before)  so here we can use the same syntax as used fot other parameters:
`    intrinsic.put("physFocalLength", principalPointX); `

Reply:
The field can not be optional in RGBDSensorParamParser.cpp, when reading the config file.
But in this function (IntrinsicParams::fromProperty) it can be optional, and for example the distortion parameters were already optional, so I did the same for this new parameter.